cdap-ui/app/cdap/components/PipelineNodeGraphs/PipelineNodeMetricsGraph.js

```diff
@@ -202,12 +322,108 @@ export default class PipelineNodeMetricsGraph extends Component {
       return <EmptyMessageContainer message={T.translate(`${PREFIX}.nodata`)} />;
     }
 
+    if (!Array.isArray(data) && typeof data === 'object') {
+      return (
+        <div className="node-metrics-single-datapoint">
+          {
+            Object.keys(data).map(key => {
+              return (
+                <span>
+                  <small>{data[key].label}</small>
+                  <span>{isNil(data[key].data) ? T.translate('commons.notAvailable') : data[key].data}</span>
+                </span>
+              );
+            })
+          }
+        </div>
+      );
+    }
     return (
       <div className="node-metrics-single-datapoint">
         {data}
       </div>
     );
   }
+
+  /*
+    We get processing time (min, max and stddev) metrics in microseconds.
+    The calculation here is
+    if time < 1000 then show it as milliseconds
+    if time > 1000 then show it in human readable duration
+
+    Process rate: Records out/Total time
+    Avg Processing Time: 1 / Process rate
+  */
+  renderProcesstimeTable = () => {
+    let totalProcessingTime, validRecords;
+    Object
+      .keys(this.state.processTimeMetrics)
+      .forEach(metric => {
+        if (metric.match(/user.*.process.time.total/)) {
+          totalProcessingTime = this.state.processTimeMetrics[metric] / 1000000;
```

Comment:
Is `totalProcessingTime` supposed to be in microseconds? (I'm assuming `this.state.processTimeMetrics[metric]` is in seconds here)